web/pgadmin/authenticate/oauth2.py

```diff
@@ -159,12 +159,16 @@ def login(self, form):
                 self.oauth2_current_client
             ]['OAUTH2_ADDITIONAL_CLAIMS']
 
-        (valid, reason) = self.__is_additional_claims_valid(profile,
-                                                            additinal_claims)
-
-        if not valid:
+        (valid_profile, reason) = self.__is_any_claims_valid(profile,
+                                                             additinal_claims)
+        
+        id_token_claims = session['oauth2_token']['userinfo']
+        (valid_idtoken, reason) = self.__is_any_claims_valid(id_token_claims,
+                                                             additinal_claims)
+
+        if not valid_profile and not valid_idtoken:
             return_msg = "The user is not authorized to login" \
-                " based on the claims in the profile." \
+                " based on the claims in your identity." \
```

Comment:
I can implement two independent responses here if required, but for simplicity I'm calling both "identity"